File: algorithms/338.swift
// https://leetcode.com/problems/counting-bits

class Solution {
    func countBits(num: Int) -> [Int] {
        var result = [0]
        while result.count < num+1 { result += result.map { $0 + 1 } }
        return Array(result[0...num])
    }
}

File: algorithms/338.py
# https://leetcode.com/problems/counting-bits


class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """

        result = [0]
        while len(result) < num + 1:
            result += list(map(lambda x: x + 1, result))
        return result[:num + 1]
